src/routing/core/messaging/sending.rs

```diff
@@ -302,22 +299,30 @@ impl Core {
             if recipient.name() == &self.node.name() {
                 handle = true;
             } else {
-                others.push(*recipient.addr());
+                others.push((*recipient.name(), *recipient.addr()));
             }
         }
 
+        let dest_info = DestInfo {
+            dest: XorName::random(), // will be updated when sending
+            dest_section_pk: *self.section_key_by_name(&others[0].0),
```

Comment:
this shall be AFTER the empty check.
otherwise will throw an `out of bound` panic, when `others` is empty (for handling case)